lib/cdo/user_helpers.rb

```diff
@@ -9,7 +9,9 @@ module UserHelpers
   def self.generate_username(queryable, name)
     prefix = name.downcase.gsub(/[^#{USERNAME_ALLOWED_CHARACTERS.source}]+/, ' ')[0..16].squish.gsub(' ', '_')
 
-    prefix = 'coder' if prefix.empty? || prefix == '_'
+    if (prefix.empty? || prefix == '')
+      prefix = 'coder' + rand(1000000).to_s
```

Comment:
maybe get a random number between 100,000 and 1,000,000 to ensure that it has some minimum number of digits so that we never randomly end up with the exact same situation we're in if we randomly get a random number < 100?